include/storage/io.hpp

```diff
@@ -144,6 +144,80 @@ class FileReader
     }
 };
 
+class FileWriter
+{
+private:
+	const std::string filename;
+	boost::filesystem::ofstream output_stream;
+
+public:
+	enum FingerprintFlag
+	{
+		GenerateFingerprint,
+		HasNoFingerprint
+	};
+
+
+	FileWriter(const std::string &filename, const FingerprintFlag flag)
+		: FileWriter(boost::filesystem::path(filename), flag)
+	{
+	}
+
+	FileWriter(const boost::filesystem::path &filename_, const FingerprintFlag flag)
+		: filename(filename_.string())
+	{
+		output_stream.open(filename_, std::ios::binary);
+		if (!output_stream)
+			throw util::exception("Error opening " + filename + ":" + std::strerror(errno));
```

Comment:
Would it make sense to have separate IO exceptions? (Check the Gist in https://github.com/Project-OSRM/osrm-backend/issues/2242#issuecomment-261223565).

(btw all those commends apply to the FileReader, too cc @ghoshkaj @danpat)